client/js/main.js

```diff
@@ -88,15 +88,19 @@ export function showOverlay(id, forced) {
 
   if(id) {
     const style = $(`#${id}`).style;
-    style.display = !forced && style.display === 'flex' ? 'none' : 'flex';
-    overlayActive = style.display === 'flex';
+    style.display = !forced && style.display !== 'none' ? 'none' : 'flex';
+    overlayActive = style.display !== 'none';
     if(forced)
       overlayActive = 'forced';
 
     //Hack to focus on the Go button for the input overlay
     if (id == 'buttonInputOverlay') {
       $('#buttonInputGo').focus();
     }
+    //Hack to set grid display for add Overlay
+    if (id == 'addOverlay' && style.display !== 'none') {
+      style.display = 'grid'
```

Comment:
Instead of doing this. In JS you can set style.display to either `'none'` or `''` and then in the css file the display listed there will be used if JS sets `''`.

Alternately we could add a class or attribute for whether the overlay is displayed. I.e. typically what's done is something like this in the css file:
```css
.overlay[hidden] {
  display: none;
}
```

With JS then setting or removing the hidden attribute.

Reply:
This requires changes for every overlay if I'm not mistaken. I'm kind of reluctant to take that on in this fairly simple PR.

Reply:
All of the overlays are of class `.overlay` so you could have two rules cover all of them:
```css
overlay {
  display: flex;
}

#addOverlay {
  display: grid;
}
```

Alternately if you go with the HTML attribute to toggle visibility,
```css
overlay[hidden] {
  display: none !important;
}

overlay {
  display: flex;
}

#addOverlay {
  display: grid;
}
```

Reply:
While this is a really clean design, I'm just not comfortable putting it into this PR. Beyond the CSS changes, which are indeed fairly easy (though they require some other changes in `room.html` and a couple of overlay css files), there is at least one other place in the code where `display: none` is tested for (this is in the handler for a click on the toolbar buttons). If this is to be done, it should be its own PR.

Reply:
Okay, instead of this override here, can you move 'flex' or 'grid' into a local var so it's not an override after setting the other style. E.g. change lines 91-92 to:

```js
const activeDisplay = id == 'addOverlay' ? 'grid' : 'flex';
style.display = !forced && style.display !== 'none' ? 'none' : activeDisplay;
overlayActive = style.display !== 'none';
```

Reply:
Sold. Still unfortunate, but cleaner.